pkg/network/setup/netstat.go

```diff
@@ -38,27 +36,32 @@ type NetStat struct {
 	// In memory cache, storing pod interface information.
 	// key is the file path, value is the contents.
 	// if key exists, then don't read directly from file.
-	podInterfaceVolatileCache PodInterfaceByVMIAndName
+	podInterfaceVolatileCache sync.Map
 }
 
 func NewNetStat(ifaceCacheFactory cache.InterfaceCacheFactory) *NetStat {
 	return &NetStat{
 		ifaceCacheFactory:         ifaceCacheFactory,
-		podInterfaceVolatileCache: PodInterfaceByVMIAndName{},
+		podInterfaceVolatileCache: sync.Map{},
 	}
 }
 
 func (c *NetStat) Teardown(vmi *v1.VirtualMachineInstance) {
-	c.podInterfaceVolatileCache.DeleteAllForVMI(vmi.UID)
+	c.podInterfaceVolatileCache.Range(func(key, value interface{}) bool {
+		if strings.Contains(key.(string), string(vmi.UID)) {
+			c.podInterfaceVolatileCache.Delete(key)
+		}
+		return true
+	})
 }
 
 func (c *NetStat) PodInterfaceVolatileDataIsCached(vmi *v1.VirtualMachineInstance, ifaceName string) bool {
-	_, exists := c.podInterfaceVolatileCache.Load(vmi.UID, ifaceName)
+	_, exists := c.podInterfaceVolatileCache.Load(fmt.Sprintf("%s/%s", vmi.UID, ifaceName))
```

Comment:
This function call is repeated several times in the file, can you please consider adding a function that generates the key?

Reply:
Why do we need to wrap ‘fmt.Sprintf’ with another function?

The only advantage will be to skip the format string, and it is short and to the point that does not require explanation.

Reply:
The advantage as you've said is to skip the format string which could be error prone if a space character slips.
Furthermore, a function call could also improve the code's readability IMO.

Reply:
Thank you